### PR TITLE
Added errors handler for forked PTY process

### PR DESCRIPTION
**Pull request by frol**
Here is the issue reproduction:

```python
import invoke
invoke.run(u'echo \xff', pty=True)
```

### CODE REVIEW

File: invoke/runners.py [hunk 1]
```diff
@@ -447,26 +447,35 @@ def start(self, command):
...
+                    # before adding it!
+                    #
+                    # Set pty window size based on what our own controlling
+                    # terminal's window size appears to be.
+                    # TODO: make subroutine?
+                    winsize = struct.pack(b'HHHH', rows, cols, 0, 0)
+                    fcntl.ioctl(sys.stdout.fileno(), termios.TIOCSWINSZ, winsize)
+                    # Use execv for bare-minimum "exec w/ variable # args"
+                    # behavior. No need for the 'p' (use PATH to find executable)
+                    # or 'e' (define a custom/overridden shell env) variants, for
```
Comment (betterdiff) [inline invoke/runners.py]: * E501 line too long (81 > 79 characters)

File: invoke/runners.py [hunk 2]
```diff
@@ -447,26 +447,35 @@ def start(self, command):
...
+                    # now.
+                    # TODO: use /bin/sh or whatever subprocess does. Only using
+                    # bash for now because that's what we have been testing
+                    # against.
+                    # TODO: also see if subprocess is using equivalent of execvp...
```
Comment (betterdiff) [inline invoke/runners.py]: * E501 line too long (83 > 79 characters)

File: invoke/runners.py [hunk 3]
```diff
@@ -447,26 +447,35 @@ def start(self, command):
...
+                    # setup/teardown involving tty.setraw, getrlimit, signal.
+                    # Ostensibly we'll want some of that eventually, but if
+                    # possible write tests - integration-level if necessary -
+                    # before adding it!
+                    #
+                    # Set pty window size based on what our own controlling
+                    # terminal's window size appears to be.
+                    # TODO: make subroutine?
+                    winsize = struct.pack(b'HHHH', rows, cols, 0, 0)
+                    fcntl.ioctl(sys.stdout.fileno(), termios.TIOCSWINSZ, winsize)
```
Comment (betterdiff) [inline invoke/runners.py]: * E501 line too long (81 > 79 characters)

File: invoke/runners.py [hunk 4]
```diff
@@ -447,26 +447,35 @@ def start(self, command):
...
+                    # Use execv for bare-minimum "exec w/ variable # args"
+                    # behavior. No need for the 'p' (use PATH to find executable)
```
Comment (betterdiff) [inline invoke/runners.py]: * E501 line too long (81 > 79 characters)

**Comment by frol** [comment]
@bitprophet I discovered this bug while improving integration tests for my #274

File: invoke/runners.py [hunk 5]
```diff
@@ -447,26 +447,41 @@ def start(self, command):
...
+                    os.execv('/bin/bash', ['/bin/bash', '-c', command])
+                except Exception:
+                    # Prevent process hanging when something wrong happened
+                    # here, for example, encoding error in `execv`
+                    import traceback
+                    traceback.print_exc()
+                    # We forcefully terminate this fork branch to simulate
+                    # no-op `execv`
+                    os._exit(1)
+
```
Comment (betterdiff) [inline invoke/runners.py]: * W293 blank line contains whitespace

**Comment by frol** [comment]
Everything is tested and Travis is happy. Please, merge.

**Comment by frol** [comment]
@bitprophet You seem to neglect PRs which target stability. This is why I decided to provide even more obvious reproduction:

```python
import invoke
try:
    invoke.run(u'echo \xff', pty=True)
except:
    pass
print("This must be printed ONCE!")
```

```
$ env LANG=C python bug_275.py
This must be printed ONCE!
This must be printed ONCE!
```

The text "This must be printed ONCE!" is printed **TWICE** because of the bug which happens when `execv` fails. In this case, it fails due to another bug (fixed in #274 (yet again, not merged for almost half a year now)):

```
  File "/tmp/env/lib/python2.7/site-packages/invoke/runners.py", line 741, in start
    os.execv('/bin/bash', ['/bin/bash', '-c', command])
TypeError: execv() arg 2 must contain only strings
```

**Comment by bitprophet** [comment]
> You seem to neglect PRs which target stability.

Casting shade on the maintainer is not good feedback. You don't seem to value volunteer time very much going by this and past comments. I get that my slow pace is frustrating, but you need to understand this is all a labor of love and try dialing back the entitlement a little bit please! :) thanks!

(And thanks also for the detailed repro! I don't think I had tried reproducing this yet but the extra details make it much clearer.)

**Comment by frol** [comment]
@bitprophet It is frustrating to see that my volunteer time (and other PRs also) is left behind, while I see new commits are made. Please, prioritize PRs reviews and either close them or merge before you proceed. Otherwise, PRs are just useless.

I have some time for open-source, but I want to make it better, not to waste my time on it.

**Comment by frol** [comment]
I have merged this into #274 since the tests failing badly otherwise.

**Pull request closed without merging**